### PR TITLE
feat(sdk,swap): [WEB-499] set max byte size limit for ccmMessage

### PR DESCRIPTION
**Pull request by jerryafr**

### CODE REVIEW

File: packages/shared/src/parsers.ts [hunk 1]
```diff
@@ -20,6 +20,13 @@ const errorMap: ZodErrorMap = (_issue, context) => ({
   message: `received: ${safeStringify(context.data)}`,
 });
 
+export const hexStringWithMaxByteSize = (maxByteSize: number) =>
+  z
+    .string()
+    .refine((val) => /^0x[0-9a-f]+$/.test(val))
```
Comment (acdibble) [inline packages/shared/src/parsers.ts:26]: if possible:
```suggestion
  hexString
```

**Review by acdibble — APPROVED** [review]

File: packages/shared/src/__tests__/parsers.test.ts [hunk 2]
```diff
@@ -87,3 +93,27 @@ describe('unsignedInteger', () => {
...
+  });
+
+  it('should reject non hex strings', () => {
+    expect(() => hexStringWithMaxByteSize(100).parse('hello')).toThrow(
+      'Invalid input',
+    );
+  });
+
+  it('should reject strings larger than max byte size', () => {
+    expect(() => hexStringWithMaxByteSize(3).parse('0xabcdef')).toThrow(
```
Comment (niklasnatter) [inline packages/shared/src/__tests__/parsers.test.ts:115]: i think i would expect this to pass as the value has 3 bytes?

File: packages/shared/src/parsers.ts (hunk 1 above)
Comment (GabrielBuragev) [inline packages/shared/src/parsers.ts:26]: Do we need to care about case-sensitivity here? 
I know that in the ethereum ecosystem, the wallet addresses are many times represented as `/^0x[0-9a-fA-F]+$/`. Users use all sorts of tools to copy these addresses and sometimes the result is, they have a checksummed address in their clipboard.
I guess it depends on what we expect to be passed as a value for the ccm but just wanted to raise the awareness on this one.

Comment (niklasnatter) [reply packages/shared/src/parsers.ts:26]: good point! im actually not sure how the backend handles this - worth a check if it accepts uppercase letters

Comment (niklasnatter) [reply packages/shared/src/parsers.ts:26]: yep sounds good - can we reuse/adjust the existing `hexString`?

Comment (jerryafr) [reply packages/shared/src/parsers.ts]: updated it to reuse `hexString` which actually is case insensitive

**Review by niklasnatter — APPROVED** [review]

**Pull request merged**